### PR TITLE
Fix sticking “Reset” option in `ToolsPanel`

### PR DESCRIPTION
**Pull request by stokesman**
## What?
Ensures the “Reset” action of tools panel’s menu items is present only when an item’s value is set.

## Why?
Currently once a panel item has had its value set the ”Reset” option becomes enabled but it gets stuck that way even if the original value has been restored. This doesn't cause much of a problem so fixing it could be described as UI/UX polish.

Further, this is extracted from a larger effort in another branch to ensure defaults of block supports are represented in the controls. It can work without these changes but the UI discrepancy is perhaps more glaring in such a context.

## How?

- Changes an effect to run anytime a panel item’s value changes and modifies the callback it triggers to accept the actual value (instead being hardcoded to `true`).

### In the Editor
1. Go to Styles > Color
2. Pick a color for one of the elements
3. Activate the same picked color swatch to clear the picked value
4. Check the options menu and note that “Reset” is not available.

It's important to clear the value by using the same swatch as before. If you actually switch back to the original value by using its swatch then “Reset” will be enabled. This is a separate issue and fixing it also depends on these changes.

### In Storybook
1. Go to ToolsPanel
2. Enter a value in one of the example fields
3. Delete the value
4. Check the options menu and note that ”Reset” is not available.

### Testing Instructions for Keyboard
Not sure it’s required. The instructions above aren’t exclusively for pointing devices. I'll add more detailed instruction if need be.

### Before

https://github.com/WordPress/gutenberg/assets/9000376/1415f27a-5933-48c6-b791-e1e167cf6d45

### After

https://github.com/WordPress/gutenberg/assets/9000376/c787ecaa-d515-485e-96de-a0cee390ee67

### CODE REVIEW

File: packages/components/src/tools-panel/tools-panel-item/hook.ts
Comment (stokesman) [inline packages/components/src/tools-panel/tools-panel-item/hook.ts:160]: The effect that triggers this runs more often now that `flagItemCustomization` get called when removing a value. That was causing `onDeselect` to be called multiple times and this change fixed that.

As an aside, I experimented with a little refactor to not call these (`onSelect`/`onDeselect`) from an effect and it seemed to work well and simplifies things a bit. I wanted to keep changes minimal so I've left it out but if that sounds good I'd happily make a PR for it.

File: packages/components/src/tools-panel/tools-panel/hook.ts
Comment (stokesman) [inline packages/components/src/tools-panel/tools-panel/hook.ts:211]: The comment seemed out of date because the callback is already being used from optional panel items and now, of course, it’s being used to also “remove” a customization flag.

**Review by stokesman — COMMENTED** [review]
🙇

**Review by tyxla — APPROVED** [review]
Looks good and tests well.  ✅ 

I'd personally add a CHANGELOG entry, but that's not a requirement.

Thanks @stokesman 🙌

**Comment by stokesman** [comment]
> I'd personally add a CHANGELOG entry,

Me too, I just tend to wait for approval and try sparing conflict fixes.

Thanks for testing and reviewing Marin ❤️.

**Pull request merged**
